Resources/Prototypes/_NF/Entities/Objects/Weapons/Guns/turrets.yml

```diff
@@ -0,0 +1,313 @@
+  name: laser turret
+  suffix: Frontier, Hostile
+  components:
+  - type: NpcFactionMember
+    factions:
+    - SimpleHostile
+
+## Magazine fed, deployable NanoTrasen turret
+- type: entity # deployed turret
+  parent:
+  - BaseWeaponTurretMagazineFed
+  id: WeaponTurretAsmgtNtDeployed
+  name: ASMGT
+  description: |-
+   An Autonomous Sentry Machine Gun Turret (ASMGT) is a magazine fed ballistic turret compatible with .20, .25, .30 and .35 cartriges. Use screwdriver to disassemble.
+  suffix: NT
+  components:
+  - type: NPCImprintingOnSpawnBehaviour
+    whitelist:
+      components:
+      - HumanoidAppearance
+  - type: NpcFactionMember
+    factions:
+    - NanoTrasen
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/magazinefed.rsi
+    drawdepth: Mobs
+    layers:
+    - state: base
+      map: ["enum.GunVisualLayers.Base"]
+    - state: mag-0
+      map: ["enum.GunVisualLayers.Mag"]
+  - type: Appearance
+  - type: MagazineVisuals
+    magState: mag
+    steps: 11
+    zeroVisible: true
+  - type: ItemSlots
+    slots:
+      gun_magazine:
+        name: Magazine
+        startingItem: null
+        insertSound: /Audio/Weapons/Guns/MagIn/smg_magin.ogg
+        ejectSound: /Audio/Weapons/Guns/MagOut/smg_magout.ogg
+        priority: 2
+        whitelist:
+          tags:
+          - MagazineAsmgt
+  - type: Construction
+    deconstructionTarget: null
+    graph: WeaponTurretAsmgtNtPackingGraph
+    node: WeaponTurretAsmgtNtPacked
+
+- type: entity # folded turret
+  parent: BaseItem
+  id: WeaponTurretAsmgtNtPacked
+  name: packed ASMGT
+  description: |-
+   A disassembled and packed Autonomous Sentry Machine Gun Turret (ASMGT).
+  suffix: NT, Packed
+  components:
+  - type: WeaponCaseInsertable
+  - type: MultiHandedItem
+  - type: SpawnAfterInteract
+    prototype: WeaponTurretAsmgtNtDeployed
+    doAfter: 5
+    removeOnInteract: true
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/magazinefed.rsi
+    state: packed
+  - type: Item
+    size: Ginormous
+    shape:
+    - 0,0,3,3
+    sprite: _NF/Objects/Weapons/Guns/Turrets/magazinefed.rsi
+
+- type: constructionGraph # deconstruction graph
+  id: WeaponTurretAsmgtNtPackingGraph
+  start: WeaponTurretAsmgtNtPacked
+  graph:
+  - node: WeaponTurretAsmgtNtPacked
+    entity: WeaponTurretAsmgtNtDeployed
+    edges:
+    - to: WeaponTurretAsmgtNtPacked
+      completed:
+      - !type:SpawnPrototype
+            prototype: WeaponTurretAsmgtNtPacked
+            amount: 1
+      - !type:EmptyAllContainers
+      - !type:DeleteEntity {}
+      steps:
+        - tool: Screwing
+          doAfter: 14
+
+## Magazine fed, deployable punk turret
+- type: entity # deployed turret
+  parent:
+  - WeaponTurretAsmgtNtDeployed
+  - UnregisteredFirearmBase
+  id: WeaponTurretAsmgtHostileUniversallyDeployed
+  suffix: Universally Hostile
+  components:
+  - type: NPCImprintingOnSpawnBehaviour
+    whitelist:
+      components:
+      - HumanoidAppearance
```

Comment:
Does this end up attacking everybody except the one who built it?  If so, interesting.

Reply:
Friends are tracked within a 3m radius on spawn, got it.  Maybe there should be a warning in the unpacked entity's description?

Reply:
Maybe there should be a warning on packed entity. Gotta think how to word it though.